pkg/api/validation/events.go

```diff
@@ -25,11 +25,20 @@ import (
 // ValidateEvent makes sure that the event makes sense.
 func ValidateEvent(event *api.Event) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
-	if event.Namespace != event.InvolvedObject.Namespace {
-		allErrs = append(allErrs, errs.NewFieldInvalid("involvedObject.namespace", event.InvolvedObject.Namespace, "namespace does not match involvedObject"))
-	}
+
 	if !util.IsDNSSubdomain(event.Namespace) {
 		allErrs = append(allErrs, errs.NewFieldInvalid("namespace", event.Namespace, ""))
 	}
+	// TODO: event.InvolvedObject is not versioned. References need to be normalizd to internal when converted from external and vice versa.
+	if event.InvolvedObject.Kind == "Minion" || event.InvolvedObject.Kind == "Node" {
```

Comment:
Well, I know there will be cases where we send an event for a type core Kube doesn't know about.  How do we handle that?  RESTMapper could maybe deal with dynamically registered types, but what if events start going before a type is registered?  I'm starting to think you can't do this validation up front, you can only do it at the point where you use it (in a UI or cli or tool that works with known events).


> On Feb 17, 2015, at 10:17 PM, Deyuan Deng <notifications@github.com> wrote:
> 
> In pkg/api/validation/events.go:
> 
> >  	if !util.IsDNSSubdomain(event.Namespace) {
> >  		allErrs = append(allErrs, errs.NewFieldInvalid("namespace", event.Namespace, ""))
> >  	}
> > +	// TODO: event.InvolvedObject is not versioned. References need to be normalizd to internal when converted from external and vice versa.
> > +	if event.InvolvedObject.Kind == "Minion" || event.InvolvedObject.Kind == "Node" {
> The latest.RESTMapper is somewhat a raw interface for handling rest mapping. It looks weired that we translate request object in rest level using restmapper, and pass restmapper to validation level to handle referenced object. I think we should do what @smarterclayton suggested, handle it before validation. But instead of filtering just for event, can we generalize it? For example, extend scopeNamer to return reference object scope, etc. Not very familiar with our rest code, but feel like this is the correct way to go.
> 
> —
> Reply to this email directly or view it on GitHub.
>